src/Symfony/Component/Validator/Constraints/BackedEnumValueValidator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * BackedEnumValueValidator validates that a backed enum case can be hydrated from a value.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class BackedEnumValueValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof BackedEnumValue) {
+            throw new UnexpectedTypeException($constraint, BackedEnumValue::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        try {
+            $enumTypeValue = $constraint->type::tryFrom($value);
+        } catch (\TypeError) {
+            $this->context->buildViolation($constraint->typeMessage)
+                ->setParameter('{{ type }}', $this->formatValue((string) (new \ReflectionEnum($constraint->type))->getBackingType()))
+                ->setCode(BackedEnumValue::INVALID_TYPE_ERROR)
+                ->addViolation();
+
+            return;
+        }
+
+        if (null === $enumTypeValue) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setParameter('{{ choices }}', $this->formatValidCases($constraint))
```

Comment:
@derrabus is right, you only need to define what is in the constraint message

Reply:
It's build in a similar way as Choice constraint.

Message : https://github.com/symfony/symfony/blob/78f4d9a1059725384ce563c4e1696b8e66c6983a/src/Symfony/Component/Validator/Constraints/Choice.php#L41

Validator : https://github.com/symfony/symfony/blob/78f4d9a1059725384ce563c4e1696b8e66c6983a/src/Symfony/Component/Validator/Constraints/ChoiceValidator.php#L101

The `value` or `choice` values are not used by default. If you override the default message and use `{{ value }}` or ``{{ choice }}` on it, it will be replaced in your new message.